packages/zowe-explorer-api/src/vscode/ZoweVsCodeExtension.ts

```diff
@@ -234,12 +236,15 @@ export class ZoweVsCodeExtension {
             });
             await (zeRegister?.getCommonApi(serviceProfile).logout ?? Logout.apimlLogout)(updSession);
 
-            const connOk = serviceProfile.profile.host === baseProfile.profile.host && serviceProfile.profile.port === baseProfile.profile.port;
-            if (connOk) {
-                await cache.updateBaseProfileFileLogout(baseProfile);
-            } else {
-                await cache.updateBaseProfileFileLogout(serviceProfile);
-            }
+            const connOk =
+                serviceProfile.profile.host === baseProfile.profile.host &&
+                serviceProfile.profile.port === baseProfile.profile.port &&
+                !serviceProfile.name.startsWith(baseProfile.name + ".");
+            await cache.updateBaseProfileFileLogout(connOk ? baseProfile : serviceProfile);
+            return true;
+        } else {
+            Gui.errorMessage(`Logout failed: No base profile found to store SSO token for profile "${serviceProfile.name}"`);
```

Comment:
Since this is the logout operation... should we change the wording a bit?
```suggestion
            Gui.errorMessage(`Logout failed: No base profile found to remove the SSO token for profile "${serviceProfile.name}"`);
```